src/charm.py

```diff
@@ -511,6 +511,10 @@ def _config_changed(self, event: HookEvent) -> None:
                 else "",
             )
         env_settings = self._create_discourse_environment_settings()
+        if self.config.get("s3_enabled"):
+            # We force assets to be uploaded to S3
+            # This should be considered as a workaround and revisited later
+            env_settings.update({"FORCE_S3_UPLOADS": "true"})
```

Comment:
I think this might be better in the `_get_s3_env` method?

Reply:
I wanted to force this only when uploading assets that way. To avoid further possible disruptions by this workaround

Reply:
But I'm okay to change its location if you think it's better there

Reply:
I went ahead and moved it to `_get_s3_env` as it is conceptually simpler